packages/editor/src/types/config.ts

```diff
@@ -18,16 +18,23 @@ export type CreateData = { componentName: string; label: string; value: string;
 
 export const isCreateData = (data: unknown): data is CreateData => typeof data === 'object' && data !== null && 'componentName' in data;
 
-type Subsection = 'General' | 'Styling' | 'Behaviour';
+type Subsection = 'General' | 'Styling' | 'Behaviour' | 'Static Options' | 'Dynamic Options';
 
 export type BaseField<ComponentProps extends DefaultComponentProps = DefaultComponentProps> = {
   subsection: Subsection;
   label?: string;
   hide?: (component: ComponentProps) => boolean;
   section?: 'Layout';
 };
+
+export type TextFieldOptions = {
+  placeholder?: string;
+  displayOnlyListTypes?: Boolean;
+};
+
 export type TextField<ComponentProps extends DefaultComponentProps = DefaultComponentProps> = BaseField<ComponentProps> & {
   type: 'text' | 'number' | 'textarea' | 'checkbox' | 'textBrowser';
+  options?: TextFieldOptions;
```

Comment:
you now have the placeholder and the displayOnlyListTypes option on all this field types.
I think it is better to create specific types with the supported options, e.g. like the select field below.